plugins/woocommerce/client/legacy/js/admin/meta-boxes-order.js

```diff
@@ -1585,11 +1588,112 @@ jQuery( function ( $ ) {
 					$('table#list-table').show();
 				},
 
-				delBefore: function( settings ) {
+				delBefore: function( settings, el ) {
+					if (typeof select2_args.ajax == 'undefined') {
+						// If the list of meta keys have already loaded, prepend the deleted key to the list if it isn't already present.
+						let meta_key = $(el).find('#meta-' + settings.data.id + '-key').val();
+						if (metakey_select.find('option[value=\'' + meta_key + '\']').length === 0) {
+							let newOption = new Option(meta_key, meta_key, false, false);
+							metakey_select.prepend(newOption);
+						}
+					}
 					settings.data.order_id = woocommerce_admin_meta_boxes.post_id;
 					settings.data.action   = 'woocommerce_order_delete_meta';
 					return settings;
-				}
+				},
+
+			});
+
+			$( '#order_custom #metakeyselect').filter( ':not(.enhanced)' ).each( function() {
+				metakey_select = $(this);
+				select2_args = {
+					allowClear: !!$(this).data('allow_clear'),
+					placeholder: $(this).data('placeholder'),
+					escapeMarkup: function (m) {
+						return m;
+					},
```

Comment:
Is this required to simply be set? I can't make heads or tails of what the purpose of this is :D